Technical Review: Add documentation for Fenced Frames (#27781)
mdn/content: files/en-us/web/html/element/fencedframe/index.md

---
title: "<fencedframe>: The Fenced Frame element"
slug: Web/HTML/Element/fencedframe
page-type: html-element
status:
  - experimental
browser-compat: html.elements.fencedframe
---

{{HTMLSidebar}}{{SeeCompatTable}}

The **`<fencedframe>`** [HTML](/en-US/docs/Web/HTML) element represents a nested {{Glossary("browsing context")}}, embedding another HTML page into the current one. `<fencedframe>`s are very similar to `<iframe>`s in form and function, except that:

- Communication cannot be shared between the `<fencedframe>` content and its embedding site.
- `<fencedframe>`s can access cross-site data, but only in a very specific set of controlled circumstances that preserve user privacy.

> - `<fencedframes>`s cannot be manipulated or have their data accessed via regular scripting (for example reading or setting the source URL). They can only have content embedded via specific APIs, which we'll talk about below.

Author: Good call. I'm going to handle this one slightly differently in my next commit; the main difference is that I've linked to the specific place on the Fenced Frame API landing page where there APIs that can set `<fencedframe>` content are listed.

- The `<fencedframe>` cannot access the embedding context's DOM, or vice versa.

> In effect, `<fencedframe>`s are a more privacy-respecting type of `<iframe>`, aimed at fulfilling different use cases that previously relied on third-party cookies or other mechanisms that were bad for privacy. See [Fenced Frame API](/en-US/docs/Web/API/Fenced_frame_API) for more details.

Reviewer: ```suggestion
The `<fencedframe>` is a type of `<iframe>` with more native privacy features built-in, addressing `<iframe>` shortcomings, such as reliance on third-party cookies and other privacy risks. See [Fenced frame API](/en-US/docs/Web/API/Fenced_frame_API) for more details.
```

Author: Addressed this in my next commit, with some wording tweaks.

## Attributes

This element includes the [global attributes](/en-US/docs/Web/HTML/Global_attributes).

- `allow`

>   - : Specifies a [Permissions Policy](/en-US/docs/Web/HTTP/Permissions_Policy) for the `<fencedframe>`, which defines what features are available to the `<fencedframe>` based on the origin of the request. Only [privacy sandbox](https://developer.chrome.com/docs/privacy-sandbox/) features can be controlled via a policy set on a fenced frame. See [Permissions policies available to fenced frames](#permissions_policies_available_to_fenced_frames) for more details.

Reviewer: The Chrome link is not a good link for this. Opt for a document that lists privacy sandbox features that can be controlled

Reviewer: we need the enumerated list here

Author: I know it's not ideal, but the reality is that it is going to be a while before we have all of the relevant features documented on MDN. To "fix" this for now, I've opted to just remove the privacy sandbox link altogether and link to the "Permissions policies available to fenced frames" section.

- `height`

>   - : The height of the fenced frame in CSS pixels. The default is `150`.

Reviewer: ```suggestion
  - : The height of the fenced frame in CSS pixels. Must be an integer without a unit. The default is `150`.
```

Reviewer: mention that html sizing is overridden by css sizing

- `width`

>   - : The width of the fenced frame in CSS pixels. The default is `300`.

Reviewer: same

## Permissions policies available to fenced frames

> Standard web features that can have their availability controlled via [Permissions Policy](/en-US/docs/Web/HTTP/Headers/Permissions-Policy) (for example, [`camera`](/en-US/docs/Web/HTTP/Headers/Permissions-Policy/camera) or [`geolocation`](/en-US/docs/Web/HTTP/Headers/Permissions-Policy/geolocation)) are **not available** within fenced frames. This is because permissions delegated from the top-level context to a fenced frame for allowing and denying features could be used as a communication channel, so constitute a privacy threat.

Reviewer: This takes two readings because the title is what is available and we're starting with what isn't, and then the "not available" is at the end of the sentence. Maybe reorder, and start the paragraph with "Because permissions delegated from top-level contexts to frames can be privacy threats, standard web features that can have their availability controlled via permissions policies are not available to `<fencedframes>`." not sure

Author: Yeah, I agree that this paragraph was a bit topsy turvy. In my next commit, I've moved the order around as suggested and gotten rid of the privacy sandbox link in the next para.

> The only features that can be controlled by a policy inside fenced frames are the [privacy sandbox](https://developer.chrome.com/docs/privacy-sandbox/) features designed to be used inside fenced frames:

Reviewer: again, link to the list... whichever page lists the attribution reporting API and will list others

Author: As mentioned in the previous paragraph, the link was removed. I'm intending to add more features to the table as they are added to MDN.

| Feature                                                                                               | Permissions policy directive |
| ----------------------------------------------------------------------------------------------------- | ---------------------------- |
| [Attribution Reporting API](https://developer.chrome.com/docs/privacy-sandbox/attribution-reporting/) | `attribution-reporting`      |

> > **Note:** More features will be added to the list and they will be better documented as more MDN docs are written on privacy sandbox features.

Reviewer: ```suggestion
> **Note:** More features will be added to the list as more MDN docs gets updated with new privacy sandbox features.
```

Author: My original text was pretty sloppy ;-) I've fixed it in a slightly different way in my next commit.

Currently these are always enabled inside fenced frames. In the future, which ones are enabled will be controllable using the `<fencedframe>` `allow` attribute. Blocking privacy sandbox features in this manner will also block the fenced frame from loading — there will be no communication channel at all.

## Focusing across fenced frame boundaries

The ability of the document's active focus to be moved across fenced frame boundaries (i.e. from an element outside the fenced frame to one inside, or vice versa) is limited. User-initiated actions such as a click or a tab can do so, as there is no fingerprinting risk there.

However, trying to traverse the boundary via an API call such as {{domxref("HTMLElement.focus()")}} is prohibited — a malicious script could use a series of such calls to leak inferred information across the boundary.

## Positioning and scaling

> As a [replaced element](/en-US/docs/Web/CSS/Replaced_element), the position, alignment, and scaling of the embedded document within the `<iframe>` element's box, can be adjusted with the {{cssxref("object-position")}} and {{cssxref("object-fit")}} properties.

Author: Note added to this section, similar to what was added to the HTMLFencedFrame.width/height pages.
